docs/src/advanced_topics/execution_trace_optimization.md

```diff
@@ -0,0 +1,44 @@
+# Execution trace optimization
+
+## Understanding cycle counts in Miden VM
+
+When we refer to "number of cycles" in most Miden VM documentation, we're specifically referring to the **stack rows** portion of the execution trace. However, the actual proving time is determined by what we call the "true number of cycles," which is the maximum of all trace segment lengths:
+
+- **Stack rows**: One row per VM operation (what `clk` outputs)
```

Comment:
I would mention that this corresponds to the System, Program decoder and Operand Stack set of columns from the [execution trace diagram.](https://0xmiden.github.io/miden-vm/design/main.html#vm-execution-trace)